documentation/sphinx/source/developer-guide.rst

```diff
@@ -756,6 +756,118 @@ If you only need to detect the *fact* of a change, and your response doesn't dep
 
 .. _developer-guide-peformance-considerations:
 
+
+Special keys
+============
+
+Keys starting with the bytes ``\xff\xff`` are called "special" keys, and they are materialized when read. :doc:`\\xff\\xff/status/json <mr-status>` is an example of a special key.
+As of api version 630, additional features have been exposed as special keys and are available to read as ranges instead of just individual keys. Additionally, the special keys are now organized into "modules".
+
+Modules
+-------
+
+A module is loosely defined as a key range where a user can expect similar behavior from reading any key in that range.
+By default, users will see a ``special_keys_no_module_found`` error if they read from a range not contained in a module.
+This is likely to be an error since the read would always return an empty set of keys and can't be interesting (maybe it was a typo?).
+Users will also (by default) see a ``special_keys_cross_module_read`` error if their read spans a module boundary.
+This is also potentially an error, since the user may be surprised by seeing the behavior of multiple modules in the same read.
+Users may opt out of these restrictions by setting the ``special_key_space_relaxed`` transaction option.
+
+Each special key that existed before api version 630 is its own module. These are
+
+#. ``\xff\xff/cluster_file_path`` Path to this database's cluster file
+#. ``\xff\xff/connection_string`` The desired contents of the cluster file. Should only be different if the client does not have permission to write to the cluster file.
```

Comment:
I might be dumb. What does the sentence `Should only be different if the client does not have permission to write to the cluster file.` mean here?

Reply:
https://apple.github.io/foundationdb/administration.html#accessing-cluster-file-information-from-a-client

Just says this key is the "contents of the cluster file", which is technically inaccurate (it's the _desired_ contents of the cluster file. It might not actually be the contents of the cluster file). I was trying to capture that subtle but potentially important point here.

Maybe we just clarify in the administration page and then link to it here?

Reply:
"Clarify in the admin page and link to it here" is a good idea.

Reply:
> Maybe we just clarify in the administration page and then link to it here?

Vote for this. The original description on the administration page is too short. to catch the subtle point.

Reply:
Done